api/features/tasks.py

```diff
@@ -1,17 +1,127 @@
+from task_processor.decorators import (
+    register_recurring_task,
+    register_task_handler,
+)
 from webhooks.constants import WEBHOOK_DATETIME_FORMAT
 from webhooks.webhooks import (
     WebhookEventType,
     call_environment_webhooks,
     call_organisation_webhooks,
 )
 
+from .constants import OVERWRITE, SKIP
 from .models import HistoricalFeatureState
 
 
+@register_recurring_task(
+    run_every=timedelta(hours=12),
+)
+def clear_stale_feature_imports_and_exports() -> None:
+    two_weeks_ago = timezone.now() - timedelta(days=14)
+    FeatureExport.objects.filter(created_date__lt=two_weeks_ago).delete()
+    FeatureImport.objects.filter(created_date__lt=two_weeks_ago).delete()
+
+
+@register_task_handler()
+def export_features_for_environment(
+    environment_id: int, tag_ids: Optional[list[int]] = None
+) -> None:
+    additional_filters = Q(
+        identity__isnull=True,
+        feature_segment__isnull=True,
+        feature_state_value__isnull=False,
+        feature__is_archived=False,
+    )
+
+    if tag_ids:
+        additional_filters &= Q(feature__tags__in=tag_ids)
+
+    environment = Environment.objects.get(id=environment_id)
+    feature_states = get_environment_flags_list(
+        environment=environment,
+        additional_filters=additional_filters,
+    )
+
+    payload = []
+    for feature_state in feature_states:
+        multivariate = []
+        for mv_option in feature_state.feature.multivariate_options.all():
+            mv_feature_state_value = (
+                feature_state.multivariate_feature_state_values.filter(
+                    multivariate_feature_option=mv_option
+                ).first()
+            )
+
+            multivariate.append(
+                {
+                    "percentage_allocation": mv_feature_state_value.percentage_allocation,
+                    "default_percentage_allocation": mv_option.default_percentage_allocation,
+                    "value": mv_option.value,
+                }
+            )
+
+        payload.append(
+            {
+                "name": feature_state.feature.name,
+                "default_enabled": feature_state.feature.default_enabled,
+                "is_server_key_only": feature_state.feature.is_server_key_only,
+                "initial_value": feature_state.feature.initial_value,
+                "value": feature_state.feature_state_value.value,
+                "enabled": feature_state.enabled,
+                "multivariate": multivariate,
+            }
+        )
+
+    FeatureExport.objects.create(
+        environment_id=environment_id, data=json.dumps(payload)
+    )
+
+
+@register_task_handler()
+def import_features_for_environment(feature_import_id: int) -> None:
+    feature_import = FeatureImport.objects.get(id=feature_import_id)
+    environment = feature_import.environment
+    input_data = json.loads(feature_import.data)
+    project = environment.project
+
+    for rec in input_data:
```

Comment:
I'm not sure what `rec` means here?

Reply:
Talked offline: rec means record, changing to another variable